ssl/ssl_local.h

```diff
@@ -1888,15 +1922,6 @@ typedef struct sigalg_lookup_st {
     int enabled;
 } SIGALG_LOOKUP;
```

Comment:
Granted, this may be possible -- but I see two reasons to keep things as-is in this PR:
1) To keep the amount of code changed to a minimum: I'm (comparatively) an "OpenSSL apprentice" and am wary to "change too much around" -- particularly in a PR of this dimension -- admittedly a point possibly moot to OpenSSL code experts.
2) I personally like the NID concept: a) It's much less prone to memory errors than strings and b) requires less memory (I'm an 80(80:)s child :) 

--> I'd very much prefer to move such change to a separate PR: OK?

Reply:
I've done a deep dive into what my comment would mean, and I agree that it's not for this PR.

Reply:
...  but, I still worry about cases like EC (key type) vs ECDSA (sig).  I haven't looked thoroughly, but I suppose that we do some sort of weird hack in places to counter that case...  will that case happen again?  I can't rule that out, so I would personally like it if there was this added field:

``` C
    /* NID for the key type.  This is often, but not always, the same as |sig| */
    int keytype;
```

This would naturally be the same NID as can be found in `SSL_CERT_LOOKUP` instances.

The question is if this should happen in a different PR or if you're willing to do this.  This should of course be reflected with an additional parameter of the TLS-SIGALG capability, which I have kinda sorta hinted at in my review of the `TLS_SIGALG_INFO` structure.